pkg/libvirttools/virtualization.go

```diff
@@ -83,6 +84,36 @@ type domainSettings struct {
 	systemUUID       *uuid.UUID
 }
 
+// Define a struct to store the device id info which will be used in pci-passthrough.
+type pciInfos struct {
+	pciHostDomain   uint
+	pciHostBus      uint
+	pciHostSlot     uint
+	pciHostFunction uint
+}
+
+// getPciInfo converts a device id to domain, bus, slot and function with uint, which
+// will be used in libvritxml.
+func getPciInfo(deviceID string) pciInfos {
+	tmp := strings.Split(deviceID, ":")
+	funTmp := strings.Split(tmp[2], ".")
+	pciHostDomain, err := strconv.ParseUint(tmp[0], 16, 0)
+	pciHostBus, err := strconv.ParseUint(tmp[1], 16, 0)
```

Comment:
You are constantly overwriting the same `err` variable so in effect only last operation will be validated through `if` clausule.
Please also consider usage of https://golang.org/pkg/fmt/#Scanf instead of multiple splits/manual parsing.

Reply:
You are right, I will reconsider it and reduce the splits and parsing. Thanks